File: data/Dtos/CartItem.cs
using data.Entities;
using Microsoft.EntityFrameworkCore.Metadata.Internal;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace data.Dtos
{
    public class CartItemRes
    {
        public int CartItemId { get; set; }
        public int CartId { get; set; }
        public Cart? Cart { get; set; }

        public int ProductId { get; set; }
        public Product? Product { get; set; }

        public decimal Quantity { get; set; }

        public decimal Rate { get; set; }

        public decimal BasePrice { get; set; }

        public int? UnitId { get; set; }
        public Unit? Unit { get; set; }
    }

    public class CartItemReqAddToCart
    {
        [Required, Range(1, int.MaxValue)]
        public int ProductId { get; set; }
        [Required, Range(0.01, double.MaxValue)]
        public decimal Quantity { get; set; }
    }
}

File: web/src/dtos/CartItem.ts
import { CartRes } from "./Cart";
import { ProductRes } from "./Product";
import { UnitRes } from "./Unit";

export interface CartItemRes {
  cartItemId?: number;
  cartId?: number;
  cart?: CartRes;
  productId?: number;
  product?: ProductRes;
  quantity?: number;
  rate?: number;
  basePrice?: number;
  unitId?: number;
  unit?: UnitRes;
}

export class CartItemReqAddToCart {
  productId?: number = 0;
  quantity?: number = 1;

  constructor(productId?: number, quantity?: number) {
    this.productId = productId;
    this.quantity = quantity;
  }
}

export class CartItemReqEdit {
  quantity?: number = 1;

  constructor(quantity?: number) {
    this.quantity = quantity;
  }
}